nncf/quantization/algorithms/min_max/torch_backend.py

```diff
@@ -0,0 +1,288 @@
+
+    @staticmethod
+    def minmax_statistic_collector(nncf_graph: NNCFGraph,
+                                   target_point: PTTargetPoint,
+                                   quantizer_config,
+                                   num_samples: int = None) -> PTMinMaxStatisticCollector:
+        return PTMinMaxAlgoBackend._statistic_collector_builder("min_max",
+                                                                nncf_graph,
+                                                                target_point,
+                                                                quantizer_config,
+                                                                num_samples)
+
+    @staticmethod
+    def mean_minmax_statistic_collector(nncf_graph: NNCFGraph,
+                                        target_point: PTTargetPoint,
+                                        quantizer_config: QuantizerConfig,
+                                        use_per_sample_stats: bool,
+                                        num_samples: int = None) -> PTMeanMinMaxStatisticCollector:
+        return PTMinMaxAlgoBackend._statistic_collector_builder("mean_min_max",
+                                                                nncf_graph,
+                                                                target_point,
+                                                                quantizer_config,
+                                                                num_samples)
+
+    @staticmethod
+    def get_weight_tensor(model: NNCFNetwork, target_point: PTTargetPoint) -> Tuple[str, torch.Tensor]:
+        node_to_op_address_maping = model.get_node_to_op_address_mapping()
+        op_address = node_to_op_address_maping[target_point.target_node_name]
+        w = model.get_module_by_scope(op_address.scope_in_model).weight
+        return target_point.target_node_name, w
+
+    @staticmethod
+    def get_weight_tensor_port_id(node: NNCFNode) -> int:
+        return None
+
+    @staticmethod
+    def get_weight_config(config: QuantizerConfig, model: NNCFNetwork) -> QuantizerConfig:
+        return deepcopy(config)
+
+    @staticmethod
+    def _get_input_scale_shape(nncf_graph: NNCFGraph,
+                               target_point: PTTargetPoint,
+                               quantization_config: QuantizerConfig):
+        is_weights = target_point.is_weight_target_point()
+        if is_weights:
+            module_node = nncf_graph.get_node_by_name(target_point.target_node_name)
+            layer_attributes = module_node.layer_attributes
+            assert isinstance(layer_attributes, WeightedLayerAttributes)
+            input_shape = layer_attributes.get_weight_shape()
+            channel_idx = layer_attributes.get_target_dim_for_compression()
+        else:
+            input_shape = nncf_graph.get_input_shape_for_insertion_point(target_point)
+            channel_idx = 1  # channel dim for activations
```

Comment:
Can we employ this `NNCFGraph` functionality in common algo?
@kshpv, @KodiaqQ, @l-bat

Reply:
Sorry for inconvenience, I mean functions `get_weight_shape`, `get_target_dim_for_compression` and `get_input_shape_for_insertion_point`. All this functions are `PTNNCFGraph` additional methods, specific for PT. My question is: should we align this functions in future and in case we should - how we will do this?